tool/minimap.re

```diff
@@ -30,8 +30,30 @@ impl MinimapState {
         let minimap = Tas::minimap_size();
         let expected_height = size * viewport.height.to_float();
         self.scale = expected_height / minimap.height.to_float();
-        let tw = minimap.width.to_float() * self.scale;
-        let th = minimap.height.to_float() * self.scale;
+        let mut tw = minimap.width.to_float() * self.scale;
+        let mut th = minimap.height.to_float() * self.scale;
+        match SETTINGS.minimap_position {
+           "TopLeft" => {
+                tw = viewport.width.to_float();
+                th = viewport.height.to_float();
+           },
+           "TopRight" => {
+                tw = minimap.width.to_float() * self.scale;
+                th = viewport.height.to_float();
+           },
+           "BottomRight" => {
+                tw = minimap.width.to_float() * self.scale;
+                th = minimap.height.to_float() * self.scale;
+           },
+           "BottomLeft" => {
+                tw = viewport.width.to_float();
+                th = minimap.height.to_float() * self.scale;
+           },
+           _ => {
+                tw = minimap.width.to_float() * self.scale;
+                th = viewport.height.to_float();
+           }
+        }
```

Comment:
Sorry for the bad names `tw` and `th`, I should have used better names. `tw` and `th` are the targeted width and targeted height of the minimap, depending on the current configured scale setting. Those don't change depending on the position of the minimap. Instead, you need to change where the minimap is located, which is calculated in the two lines below this code block.